### ISSUE TITLE
Add custom extensions as in StackEdit

### ISSUE THREAD

**Issue by benweet** [classeur/classeur#2]
The goal is to have the ability to customize:
- [ ] the markdown rendering mechanism and the HTML export (possibly with the use of [mustache functions](https://github.com/janl/mustache.js/#functions) in the templates)

**Comment by Swader** [comment]
Ah, this is basically what I wanted to suggest when I pinged you on Twitter.

We use StackEdit rather heavily for editor work at SitePoint.com, but we often have to do various shenanigans to make the HTML export compatible with our WordPress back end (custom classes for syntax highlights, and special tags like `[csharp][/csharp]` etc). If there were a way to target processed tokens like "code block" or "fenced code block with PHP as language attribute" in the HTML export and apply custom HTML to them, that would solve our problems and allow us to make the editor the official one for writing and editing operations. This would let us transition to real time collaboration on Classeur, too, rather than use Github like we do now.

So basically, the ability to turn code blocks into `[something]{{codeblocks[0].content}}[/something]` rather than a soup of HTML tags like it's currently done, would be a win.

**Comment by benweet** [comment]
Now you can add your own helpers in Settings>Handlebars helpers.

So for instance, if you want to change a tag in the export, you can add a helper like this:

``` js
Handlebars.registerHelper('transform', function(options) {
    var result = options.fn(this);
    return new Handlebars.SafeString(
        result.replace(/<pre[^>]*>/g, '<pre class="prettyprint">')
    );
});
```

And then, use this template:

```
{{#transform}}{{{file.content.html}}}{{/transform}}
```
